cypress-io/cypress
# feat: create from React component UI

- Closes #24773

### User facing changelog


n/a (this can go in the feature branch PR)

### Additional details


This PR adds the base UI for Create from React component.

### Steps to test


- Open CT with a React project (that doesn't use a custom dev server - we don't support this yet)
- Click New Spec button in the top-right corner
- Select create from component
- See the list of files that matches the file glob (excluding config files and Cypress-related files)
- Expand some of the rows
- See that if there are no React components exported from a given file that you see "No components found" in the expanded content
- See that if there are React components exported from a file that you see those components listed in the expanded content
- Verify that clicking on the component generates a spec file that correctly mounts the component

Verify that the design matches [Figma](https://www.figma.com/file/gq6zQUQ1bx2uqfnF64lwX9/Create-from-React-Component?node-id=84%3A9323&t=9HnmHXvRVAef5cxm-0) (minus the "choose a file name for this spec" which will be added with #25005)

### How has the user experience changed?


https://www.loom.com/share/bfe89f3314f5441f85d5ff97b9667140

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/4907
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### Outstanding Items

- [x] Exclude spec and config files from file chooser results
~- [ ] Add screen to allow the user to choose spec path and name~ https://github.com/cypress-io/cypress/issues/25005
- [x] The list of files is getting cut off at the bottom
